src/core/Akka/Actor/ActorCell.DeathWatch.cs

```diff
@@ -238,7 +240,8 @@ protected void AddWatcher(IActorRef watchee, IActorRef watcher)
                 {
                     _state = _state.AddWatchedBy(watcher);
 
-                    if (System.Settings.DebugLifecycle) Publish(new Debug(Self.Path.ToString(), Actor.GetType(), string.Format("now watched by {0}", watcher)));
+                    if (System.Settings.DebugLifecycle) Publish(new Debug(Self.Path.ToString(), Actor?.GetType(),
```

Comment:
`Actor` can be `null` - should never happen, but possible. The logging system is built to handle that though.

Reply:
You'll see this pattern show up several times throughout the review - not going to comment on each one individually, but that is what's going on here.